File: app/src/main/java/com/example/popularmovies/utils/DateUtils.java
package com.example.popularmovies.utils;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class DateUtils {

    public static String formatDate(String dateString) {
        SimpleDateFormat originalFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.getDefault());
        SimpleDateFormat resultFormat = new SimpleDateFormat("d MMM yyyy", Locale.getDefault());
        try {
            Date date = originalFormat.parse(dateString);
            return resultFormat.format(date);
        } catch (ParseException err) {
            return dateString;
        }
    }
}

File: app/src/main/java/com/recursia/popularmovies/utils/DateUtils.kt
package com.recursia.popularmovies.utils

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

object DateUtils {

    fun formatDate(dateString: String): String {
        val originalFormat = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
        val resultFormat = SimpleDateFormat("d MMM yyyy", Locale.getDefault())
        return try {
            val date = originalFormat.parse(dateString)
            resultFormat.format(date)
        } catch (err: ParseException) {
            dateString
        }
    }
}
